wurst/_items/MageMasher.wurst

```diff
@@ -0,0 +1,58 @@
+package MageMasher
+
+import AbilityObjEditing
+import ObjectIdGenerator
+import ClosureEvents
+import DamageType
+import ID
+import IttDummyCaster
+import Orders
+import StandardTextTags
+import LinkedList
+
+let MANABURN_ID = compiletime(ABIL_ID_GEN.next())
+let DRAIN_AMOUNT = 6.
+
+LinkedList<int> MASHER_TARGET_CLASSES = asList(
+                            UNIT_MAGE,
+                            UNIT_MAGE_1,
+                            UNIT_ELEMENTALIST,
+                            UNIT_ELEMENTALIST_1,
+                            UNIT_ELEMENTALIST_NEW,
+                            UNIT_HYPNOTIST,
+                            UNIT_DEMENTIA_MASTER,
+                            UNIT_PRIEST,
+                            UNIT_BOOSTER,
+                            UNIT_BOOSTER_1,
+                            UNIT_MASTER_HEALER,
+                            UNIT_MASTER_HEALER_1,
+                            UNIT_SAGE)
+
+@compiletime function createDummyManaburn()
+    new AbilityDefinitionManaBurndemon(MANABURN_ID)
+        ..setAnimationNames("")
+        ..setHeroAbility(false)
+        ..setItemAbility(false)
+        ..setLevels(1)
+        ..presetCooldown(lvl -> 0)
+        ..presetBoltDelay(lvl -> 0.15)
+        ..presetBoltLifetime(lvl -> 0.4)
+        ..presetCastRange(lvl -> 99999)
+        ..presetCastingTime(lvl -> 0)
+        ..presetMaxManaDrained(lvl -> DRAIN_AMOUNT)
+
+function mageMasherHit(unit attacker, unit target)
+    let owner = attacker.getOwner()
+    if isValidMasherTarget(owner, target)
+        new IttDummyCaster(0.45).castTarget(owner, MANABURN_ID, 1, Orders.manaburn, target, attacker.getPos())
+
+function isValidMasherTarget(player attackOwner, unit target) returns bool
+    return target.getOwner().isEnemyOf(attackOwner)
+        and target.isAlive()
+        and MASHER_TARGET_CLASSES.has(target.getTypeId())
+
+init
+    EventListener.add(EVENT_UNIT_DAMAGED) ->
```

Comment:
Is there not a way to use an `EVENT_UNIT_DAMAGE` event for detecting when abilities like depression land?

Reply:
Yes, but for depres that was deemed unnecessary and quite a bit of work just to apply the slow with correct timing.